src/fparser/tests/test_Fortran2003.py

```diff
@@ -2539,27 +2539,50 @@ def test_Io_Unit(): # R901
         assert isinstance(a, Name),`a`
         assert_equal(str(a),'a')
 
-def test_Read_Stmt(): # R910
-    cls = Read_Stmt
-    a = cls('read(123)')
-    assert isinstance(a, cls),`a`
-    assert_equal(str(a), 'READ(UNIT = 123)')
-
-    a = cls('read(123) a')
-    assert_equal(str(a), 'READ(UNIT = 123) a')
-    a = cls('read(123) a(  2)')
-    assert_equal(str(a), 'READ(UNIT = 123) a(2)')
 
-    a = cls('read*, a(  2), b')
-    assert_equal(str(a), 'READ *, a(2), b')
-    
-def test_Write_Stmt(): # R911
+def test_read_stmt(): # R910
+    ''' Check that we successfully parse various forms of READ statement '''
+    rcls = Read_Stmt
+    inst = rcls('read(123)')
+    assert isinstance(inst, rcls), `inst`
+    assert str(inst) == 'READ(123)'
+
+    inst = rcls('read(123) a')
+    assert str(inst) == 'READ(123) a'
+    inst = rcls('read(123) a(  2)')
+    assert str(inst) == 'READ(123) a(2)'
+
+    inst = rcls('read*, a(  2), b')
+    assert str(inst) == 'READ *, a(2), b'
+    # With format specified by label number
+    inst = rcls("READ 13, a(2)")
+    assert str(inst) == 'READ 13, a(2)'
+
+    # If there is no preceding "FMT=" or "NML=" then there is no way of
+    # knowing whether the second argument is a format string or a namelist
+    # without determining the actual type of the argument.
+    inst = rcls('read(123, a_namelist)')
+    assert str(inst) == "READ(123, a_namelist)"
+
+
+def test_write_stmt(): # R911
+    ''' Tests for various forms of Write statement '''
+    wcls = Write_Stmt
+    inst = wcls('write (123)"hey"')
+    assert isinstance(inst, wcls), `inst`
+    assert str(inst) == 'WRITE(123) "hey"'
+    assert (repr(inst) == "Write_Stmt(Io_Control_Spec_List(',', "
```

Comment:
Hmm. I can't remember. I guess it showed-up an issue that I then fixed. It might  be good to add it for new tests. I'll add it to those changed/introduced in this issue.